components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -18102,6 +18289,163 @@ public void addOperationPolicyMapping(Set<URITemplate> uriTemplates) throws APIM
         }
     }
 
+    public APIEndpointInfo getAPIEndpoint(String apiUUID, String endpointUUID) throws APIManagementException {
+        APIEndpointInfo apiEndpoint;
+        String sql = SQLConstants.APIEndpointsSQLConstants.GET_API_ENDPOINT_BY_API_UUID_AND_ENDPOINT_UUID;
+        try (Connection conn = APIMgtDBUtil.getConnection()) {
+            int apiId = getAPIID(apiUUID, conn);
+            PreparedStatement ps = conn.prepareStatement(sql);
+            ps.setInt(1, apiId);
+            ps.setString(2, endpointUUID);
+            try (ResultSet rs = ps.executeQuery()) {
+                if (rs.next()) {
+                    apiEndpoint = new APIEndpointInfo();
+                    apiEndpoint.setEndpointUuid(rs.getString("ENDPOINT_UUID"));
+                    apiEndpoint.setEndpointName(rs.getString("ENDPOINT_NAME"));
+                    apiEndpoint.setEndpointType(rs.getString("ENDPOINT_TYPE"));
+                    apiEndpoint.setEndpointConfig(fromBAtoEndpointConfigMap(
+                            rs.getBinaryStream("ENDPOINT_CONFIG")));
+                    apiEndpoint.setOrganization(rs.getString("ORGANIZATION"));
+                    return apiEndpoint;
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Error occurred while fetching data: " + e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public void deleteAPIEndpointByEndpointId(String endpointUuid) throws APIManagementException {
```

Comment:
Let's add javadoc comments to all public methods

Reply:
ok